### ISSUE TITLE
Class loading breakage: Will not be able to instrument JDK classes

### ISSUE THREAD

**Issue by paulwbill** [google/allocation-instrumenter#13]
I did everything as described on your Getting Started page.

$ uname -a
**Linux** sweden 3.5.7-03050732-generic #201403120653 SMP Wed Mar 12 10:54:56 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
$ javac -version
javac 1.8.0_45
$ java -version
java version "**1.8.0_45**"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

$ javac -classpath allocation-jdk8.jar Test.java
$ java -javaagent:allocation-jdk8.jar Test
**Class loading breakage: Will not be able to instrument JDK classes**

I downloaded the latest java-allocation-instrumenter-3.0.jar and renamed to allocation-jdk8.jar.

Interestingly enough, this was working very fine with the previous allocation.jar for JDK7. But now I need to make it work for JDK8 as well.

I am lost here, can you give me some help.

Thanks!

**Comment by jhmanson** [comment]
Keep it named java-allocation-instrumenter-3.0.jar.  The JAR file contains metadata that makes the loading JVM put it on the bootclasspath, which is required to instrument JDK classes.  The metadata has the name of the JAR file.

Alternatively, you can build from scratch and change <Boot-Class-Path> in pom.xml to be whatever you want it to be.

**Comment by mzhaom** [comment]
Looking at this issue again, I don't think it's a good idea to try to simplify the running process by hard coding the bootstrap classpath in the manifest file, because it's assuming the jar file is in the current working directory with a fixed name, which imposes unwritten constraints to the user and it's hard to follow. I would suggest just removing this file from the generated manifest file and update the document to let user specify -Xbootclasspath/a:allocation-instrumenter.jar just like specifying path in the -javaagent flag.

**Comment by mzhaom** [comment]
s/removing this file/removing this line/

**Comment by jhmanson** [comment]
There is nothing stopping the user from renaming it and adding the bootclasspath flag if they want to do so.  This just provides them a way *not* to do so.
